system/ee/EllisLab/ExpressionEngine/Cli/Cli.php

```diff
@@ -0,0 +1,376 @@
+    {
+        $help = new Help(new OptionFactory);
+
+        $help->setSummary($this->summary)
+                ->setDescr($this->description)
+                ->setUsage($this->usage)
+                ->setOptions($this->commandOptions);
+
+        $this->output->outln($help->getHelp($this->name));
+
+        exit();
+    }
+
+    /**
+     * fail the command and die
+     * @param  string $message
+     * @return null
+     */
+    public function fail($message = null)
+    {
+        if ($message) {
+            $this->output->errln("<<red>>{$message}<<reset>>");
+        }
+
+        exit(Status::FAILURE);
+    }
+
+    /**
+     * complete the command and die
+     * @param  string $message
+     * @return null
+     */
+    public function complete($message = null)
+    {
+        if ($message) {
+            $this->output->outln("<<green>>{$message}<<reset>>");
+        }
+
+        exit(Status::SUCCESS);
+    }
+
+    /**
+     * write info to the console
+     * @param  string
+     * @return null
+     */
+    public function write($message = null)
+    {
+        $this->output->outln($message);
+    }
+
+    /**
+     * write info to the console
+     * @param  string
+     * @return null
+     */
+    public function info($message = null)
+    {
+        $this->output->outln("<<green>>{$message}<<reset>>");
+    }
+
+    /**
+     * write error info to the console
+     * @param  string
+     * @return null
+     */
+    public function error($message = null)
+    {
+        $this->output->outln("<<red>>{$message}<<reset>>");
+    }
+
+    /**
+     * Ask question and get input
+     * @param  string $question
+     * @return mixed
+     */
+    public function ask($question, $default = '')
+    {
+        $this->output->out($question . ' ');
+
+        $result = $this->input->in();
+
+        return $result ?? $default;
+    }
+
+    /**
+     * Ask question and get boolean answer
+     * @param  string $question
+     * @return bool
+     */
+    public function confirm($question, $default = false)
+    {
+        $choices = '(yes/no)';
+
+        $defaultText = $default ? 'yes' : 'no';
+
+        $defaultChoice = "<<white>>[<<yellow>>{$defaultText}<<white>>]<<reset>>";
+
+        $this->output->outln("<<green>>{$question} {$choices} {$defaultChoice}");
+
+        $answer = $this->input->in();
+
+        if (is_bool($answer)) {
+            return $answer;
+        }
+
+        $confirmationRegex = '/^y/i';
+
+        $answerIsTrue = (bool) preg_match($confirmationRegex, $answer);
+
+        if ($default === false) {
+            return $answer && $answerIsTrue;
+        }
+
+        return '' === $answer || $answerIsTrue;
+    }
+
+    /**
+     * check if command is defined among the addons
+     * @param  string $command [defaults to whatever command was called]
+     * @return bool
+     */
+    protected function commandExists($command = null)
+    {
+        $commandToParse = $command ?? $this->commandCalled;
```

Comment:
Null-coalescing operator needs to be refactored to PHP 5.6 compatible.

Reply:
Fixed on latest commits.